modules/OFConnectionManager2/module/src/bundle.c

```diff
@@ -231,14 +280,34 @@ ind_cxn_bundle_add_handle(connection_t *cxn, of_object_t *obj)
         return;
     }
 
-    if (bundle->count == bundle->allocated) {
+    /* find the right subbundle */
+    int idx;
+    if (bundle->subbundle_count == 1) {
+        idx = 0;
+    } else {
+        of_object_storage_t obj_storage;
+        of_object_t *obj = parse_message(data.data, &obj_storage);
+        if (obj->object_id == OF_BARRIER_REQUEST) {
+            /* barriers are placed in the last subbundle */
+            idx = num_subbundles_per_bundle-1;
+        } else {
+            idx = subbundle_designator(obj);
```

Comment:
subbundle_designator() should return subbundle id which is < num_subbundles_per_bundle-1.
If is returns greater than that, then subbundle id should be num_subbundles_per_bundle where barriers are stored. If I register 3 subbundles and return 10 from designator then this will not work.